owncloudApp/src/main/java/com/owncloud/android/presentation/accounts/ManageAccountsDialogFragment.kt

```diff
@@ -0,0 +1,267 @@
+        closeButton.setOnClickListener {
+            dismiss()
+        }
+
+        val dialog = builder.create()
+        dialog.window?.setBackgroundDrawableResource(R.color.transparent)
+
+        return dialog
+    }
+
+    override fun removeAccount(account: Account) {
+        dismiss()
+        val hasAccountAttachedCameraUploads = manageAccountsViewModel.hasCameraUploadsAttached(account.name)
+        val dialog = AlertDialog.Builder(requireContext())
+            .setMessage(getString(
+                if (hasAccountAttachedCameraUploads) R.string.confirmation_remove_account_alert_camera_uploads
+                else R.string.confirmation_remove_account_alert, account.name)
+            )
+            .setPositiveButton(getString(R.string.common_yes)) { _, _ ->
+                val accountManager = AccountManager.get(MainApp.appContext)
+                accountManager.removeAccount(account, this, null)
+                if (manageAccountsViewModel.getLoggedAccounts().size > 1) {
+                    show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
+                }
+            }
+            .setNegativeButton(getString(R.string.common_no)) { _, _ ->
+                show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
+            }
+            .create()
+        dialog.avoidScreenshotsIfNeeded()
+        dialog.show()
+    }
+
+    override fun cleanAccountLocalStorage(account: Account) {
+        dismiss()
+        val dialog = AlertDialog.Builder(requireContext())
+            .setTitle(getString(R.string.clean_data_account_title))
+            .setIcon(R.drawable.ic_warning)
+            .setMessage(getString(R.string.clean_data_account_message, account.name))
+            .setPositiveButton(getString(R.string.clean_data_account_button_yes)) { _, _ ->
+                show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
+                manageAccountsViewModel.cleanAccountLocalStorage(account.name)
+            }
+            .setNegativeButton(R.string.drawer_close) { _, _ ->
+                show(supportFragmentManager, MANAGE_ACCOUNTS_DIALOG)
+            }
+            .create()
+        dialog.avoidScreenshotsIfNeeded()
+        dialog.show()
+    }
+
+    override fun createAccount() {
+        val accountManager = AccountManager.get(MainApp.appContext)
+        accountManager.addAccount(
+            MainApp.accountType,
+            null,
+            null,
+            null,
+            parentActivity,
+            { future: AccountManagerFuture<Bundle>? ->
+                if (future != null) {
+                    try {
+                        val result = future.result
+                        val name = result.getString(AccountManager.KEY_ACCOUNT_NAME)
+                        val newAccount = AccountUtils.getOwnCloudAccountByName(parentActivity.applicationContext, name)
+                        changeToAccountContext(newAccount)
+                    } catch (e: OperationCanceledException) {
```

Comment:
I don't think so, since the only exception relevant here is `OperationCanceledException`, which happens when pressing back in the login view. In case any other exception happens, at least catch it, but no need to log different messages for them